src/magic_folder/tahoe_client.py

```diff
@@ -338,9 +343,16 @@ def download_capability(self, cap):
 
         :returns: bytes
         """
-        get_uri = self.url.child(u"uri").replace(
-            query=[(u"uri", cap.decode("ascii"))],
-        )
+        # Visiting /uri with a directory-capability causes Tahoe to
+        # render an HTML page .. instead adding ?t=json tells it to
+        # send JSON instead. However, using ?t=json for non-directory
+        # capabilities means the content is *not* downloaded .. so,
+        # "t=json" is only added for directory-capabilities.
+        query_args = [(u"uri", cap.decode("ascii"))]
+        if is_directory_cap(cap):
+            query_args.append((u"t", u"json"))
```

Comment:
I'm not convinced that we should be inspecting the cap to determine whether we should request the json format or not. My impression is, that every caller should know whether they are requesting a directory or not, so we could have a different function for handling directories, as compared to files (and that function could even load the returned json and maybe(??) parse it).

I'm not opposed to checking that we are requesting a file before making a call without `?t=json`, but I think we should error out if that is not the case (and should maybe check for `IFileURI` instead of `IDirectoryURI`).

I'm not sure if we want to separate getting file metadata from directory listing or not; they use the same API request, but return (somewhat) different data.

Reply:
So you want like `download_directory_capability` versus `download_file_capability` or so...?

Reply:
We really are downloading a capability, though (in both cases). "A directory" in Tahoe is just a text file that contains some JSON describing the directory structure. But, sure, we could leave that out (it should be clear from the arg-name anyway).

Reply:
(Maybe the only real change we need here is to error-out from `download_capability` if its a directory-cap .. and maybe rename the method, too)...?

Reply:
This PR I think was motivated from some download changes I was making and I saw HTML getting spit out of `download_capability()` .. so a (proper) error would be just as good (since we already have a `list_directory` that uses `/uri?t=json`)

Reply:
> We really are downloading a capability,

We are downloading what a capability is pointing at :upside_down_face:; we already have the capability. You wouldn't say `download_file_name`, this feels equivalent.

Reply:
> This PR I think was motivated from some Downloader changes I was making and I saw HTML getting spit out of `download_capability()` .. so a (proper) error would be just as good (since we already have a `list_directory` that uses `/uri?t=json`)

That makes sense.